eralchemy2/main.py

```diff
@@ -226,21 +227,27 @@ def filter_resources(
     _tables = copy.deepcopy(tables)
     _relationships = copy.deepcopy(relationships)
 
-    include_tables = include_tables or [t.name for t in _tables]
+    include_tables = re.compile(
+        "|".join(f"({name})" for name in (include_tables or [t.name for t in _tables]))
+    )
     include_columns = include_columns or [c.name for t in _tables for c in t.columns]
-    exclude_tables = exclude_tables or list()
+    exclude_tables = re.compile(
+        "|".join(f"({name})" for name in (exclude_tables or [r"\?\?\?"]))
+    )
     exclude_columns = exclude_columns or list()
 
     _tables = [
-        t for t in _tables if t.name not in exclude_tables and t.name in include_tables
+        t
+        for t in _tables
+        if not exclude_tables.search(t.name) and include_tables.search(t.name)
     ]
     _relationships = [
         r
         for r in _relationships
-        if r.right_col not in exclude_tables
-        and r.left_col not in exclude_tables
-        and r.right_col in include_tables
-        and r.left_col in include_tables
+        if not exclude_tables.search(r.right_col)
```

Comment:
`exclude_table = re.compile("parent")`

This matches also:
`exclude_table.search("parent_table")`

I think we should use [re.fullmatch](https://docs.python.org/3/library/re.html#re.fullmatch) instead, to be consistent with the prior behavior?